src/cypher/RateLimitedAllowance.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.19;
+
+import {Ownable} from "@openzeppelin-contracts/contracts/access/Ownable.sol";
+import {Pausable} from "@openzeppelin-contracts/contracts/security/Pausable.sol";
+
+import {RateLimitedLibrary, RateLimit} from "@zelt/src/lib/RateLimitedLibrary.sol";
+import {RateLimitCommonLibrary} from "@zelt/src/lib/RateLimitCommonLibrary.sol";
+
+abstract contract RateLimitedAllowance is Pausable, Ownable {
+    using RateLimitedLibrary for RateLimit;
+    using RateLimitCommonLibrary for RateLimit;
+
+    event Approved(
+        address indexed token,
+        address indexed owner,
+        uint128 rateLimitPerSecond,
+        uint128 bufferCap
+    );
+
+    event SpenderChanged(address newSpender);
+
+    address public spender;
+
+    mapping(address owner => mapping(address token => RateLimit))
+        public limitedAllowance;
+
+    constructor(address owner, address _spender) Ownable() {
+        spender = _spender;
+        _transferOwnership(owner);
+    }
+
+    function approve(
+        address token,
+        uint128 rateLimitPerSecond,
+        uint128 bufferCap
+    ) external {
```

Comment:
We can use a Morpho bundler so I don't think we need a batch approve. Also I don't think your average user will want to approve more than 1 token for auto-load, this would be a more advanced user, and in practice you'd have to select which tokens somehow, so it seems like an edge case to have more than 1 token allowance open at a time.